packages/core/src/telemetry/service.test.ts

```diff
@@ -1,23 +1,122 @@
-import { afterEach, beforeEach, expect, test } from "vitest";
+import * as fs from "fs";
+import os from "os";
+import * as process from "process";
+import { afterAll, afterEach, beforeAll, expect, test, vi } from "vitest";
+import createFetchMock from "vitest-fetch-mock";
 
 import { buildOptions } from "@/config/options";
 import { TelemetryService } from "@/telemetry/service";
 
-let originalEnv: NodeJS.ProcessEnv;
+const cpus = os.cpus() || [];
 
-beforeEach(() => {
-  originalEnv = { ...process.env };
+const options = buildOptions({
+  cliOptions: { configFile: "", rootDir: "" },
+});
+
+const metadata = {
+  systemPlatform: os.platform(),
+  systemRelease: os.release(),
+  systemArchitecture: os.arch(),
+  cpuCount: cpus.length,
+  cpuModel: cpus.length ? cpus[0].model : null,
+  cpuSpeed: cpus.length ? cpus[0].speed : null,
+  memoryInMb: Math.trunc(os.totalmem() / Math.pow(1024, 2)),
+};
```

Comment:
Don't think it's necessary to do a full object match here - fine to just check if the keys are present.